### DISCUSSION TITLE
Skin Layout Editor Customisation and controls

### DISCUSSION THREAD

**Discussion by RaphiOsu** [ppy/osu#20043] [Ideas]
After testing the skin layout editor I have some suggestions to improve the overall feel of it.

When adding multiple elements on top of each other you cannot access the 'bottom' element without first moving the 'top' element, this can become an issue when you have multiple elements but need to adjust only one. A suggestion for this is to create a locking mechanic to stop elements from moving once locked and to also add a layer system to manually decide the order you would see them in, this layer mechanic would also be useful for adding elements behind leaderboards and select menus in Song Select and behind the key overlay in Gameplay

When placing skinnable files into the editor moving them precisely is almost impossible since there are no controls for it and moving it by mouse has about a 10 pixel delay before the element actually starts to move, adding more precise adjustment controls like holding shift to quickly get perfect rotation amounts (90 degrees as an example), arrow keys to move elements pixel perfectly, shift + arrow keys to move elements by 10 or so pixels at a time and holding shift while resizing an element to always scale at the correct aspect ratio would be good improvements.

Adding skinnable input buttons and being able to position them where ever on the editor would be nice to see, this would also allow the input buttons to be moved without having to place elements above or below them. Relating to this issue is the playback and visual settings getting pushed down if there is a big enough element set to TopRight Anchor and TopRight origin, this causing them to go off screen sometimes. Having these settings in fixed positions or being able to move them might be possible solutions for this.

**Comment by peppy** [comment]
Yet, there's lots of similar feedback if you search discussions. We'll get to doing a serious UX pass on the skin editor in the future.
